py/surveyqa/core.py

```diff
@@ -33,11 +35,16 @@ def change_range(i):
 
     exposures["HOURANGLE"] = [change_range(i) for i in exposures["HOURANGLE"]]
 
+    if nights is not None:
+        nights = [str(i) for i in nights]
+        exposures = exposures[[x in nights for x in exposures['NIGHT']]]
```

Comment:
Interesting.  This case trims the exposure list to only have those nights, and thus the summary would only have those nights.  That is useful for rapid debugging tests, but the original use case I was considering was to update the summary for *all* nights, while only generating the per-night pages for the list of input nights.  Maybe we need a third option to disambiguate these cases.  Thinking...

Reply:
Maybe I can change the boolean parameter for the summary page to a integer parameter, so if 
=0 : don't make summary page 
=1: make summary page on subset provided (if no subset provided, make summary page on all nights)
=2: make summary page on all nights

I can implement this, and if you don't like it, we can revert it back to the previous commit.

Reply:
I really like that idea, and I like that you clearly documented the options and considered the cases where the user specifies `show_summary=1` (summary on subset) but doesn't provide a subset of nights, and explicitly raise an exception if they provide an invalid value.

Let's take this one step further and use human friendly strings, which will make the code that calls it clearer (because you don't have to lookup the docstring for `surveyqa.core.makeplots()` to understand what the input values mean).  Suggestion:
  * show_summary = 'no': don't make summary page
  * show_summary = 'subset': make summary page on subset of nights provided (or all nights if no nights subset is given)
  * show_summary = 'all': make summary page on all nights
  * show_summary = anything else: raise ValueError (i.e. a more specific exception than "Exception", used for the specific case of an input having the right type but an invalid value).

What do you think?